app/services/hashtag_autocomplete_service.rb

```diff
@@ -181,15 +201,71 @@ def search(term, types_in_priority_order, limit: 5)
     # For example, if there is a category with the slug #general and a tag
     # with the slug #general, then the tag will have its ref changed to #general::tag
     top_ranked_type = slugs_by_type.keys.first
-    results.each do |hashtag_item|
+    limited_results.each do |hashtag_item|
       next if hashtag_item.type == top_ranked_type
 
-      other_slugs = results.reject { |r| r.type === hashtag_item.type }.map(&:slug)
+      other_slugs = limited_results.reject { |r| r.type === hashtag_item.type }.map(&:slug)
       if other_slugs.include?(hashtag_item.slug)
         hashtag_item.ref = "#{hashtag_item.slug}::#{hashtag_item.type}"
       end
     end
 
-    results.take(limit)
+    limited_results.take(limit)
+  end
+
+  # TODO (martin) Remove this once plugins are not relying on the old lookup
+  # behavior via HashtagsController when enable_experimental_hashtag_autocomplete is removed
+  def lookup_old(slugs)
```

Comment:
This method can be completely ignored, it's just the old lookup code and has not changed in any way.